tests/snowpark/test_function.py

```diff
@@ -240,29 +240,71 @@ def test_deploy_function_needs_update_because_handler_changes(
     assert result.exit_code == 0, result.output
     assert json.loads(result.output) == [
         {
-            "object": "func1(a string default 'default value', b variant)",
+            "object": "MOCKDATABASE.MOCKSCHEMA.FUNC1(a string default 'default value', b variant)",
             "status": "definition updated",
             "type": "function",
         }
     ]
     assert queries == [
-        "create stage if not exists dev_deployment comment='deployments managed by snowcli'",
-        f"put file://{Path(project_dir).resolve()}/app.zip @dev_deployment/my_snowpark_project"
+        "create stage if not exists MOCKDATABASE.MOCKSCHEMA.DEV_DEPLOYMENT comment='deployments managed by snowcli'",
+        f"put file://{Path(project_dir).resolve()}/app.zip @MOCKDATABASE.MOCKSCHEMA.DEV_DEPLOYMENT/my_snowpark_project"
         f" auto_compress=false parallel=4 overwrite=True",
         dedent(
             """\
-            create or replace function func1(a string default 'default value', b variant)
+            create or replace function MOCKDATABASE.MOCKSCHEMA.FUNC1(a string default 'default value', b variant)
             returns string
             language python
             runtime_version=3.10
-            imports=('@dev_deployment/my_snowpark_project/app.zip')
+            imports=('@MOCKDATABASE.MOCKSCHEMA.DEV_DEPLOYMENT/my_snowpark_project/app.zip')
             handler='app.func1_handler'
             packages=('foo=1.2.3','bar>=3.0.0')
             """
         ).strip(),
     ]
 
 
+@mock.patch("snowflake.connector.connect")
+@mock.patch("snowflake.cli.plugins.snowpark.commands.ObjectManager.describe")
+@mock.patch("snowflake.cli.plugins.snowpark.commands.ObjectManager.show")
+def test_deploy_procedure_fully_qualified_name(
+    mock_om_show,
+    mock_om_describe,
+    mock_conn,
+    runner,
+    mock_ctx,
+    project_directory,
+    alter_snowflake_yml,
+    snapshot,
+):
+    mock_om_describe.side_effect = [
+        ProgrammingError("does not exist or not authorized"),
+    ] * 100
```

Comment:
`* 100` is this needed?

Reply:
Fixed as `number_of_functions_in_project = 6` :D